extensions/ql-vscode/src/upgrades.ts

```diff
@@ -150,7 +150,11 @@ export async function upgradeDatabase(
 
   try {
     qs.logger.log('Running the following database upgrade:');
-    qs.logger.log(compileUpgradeResult.compiledUpgrades.scripts.map(s => s.description.description).join('\n'));
+    if (compileUpgradeResult.compiledUpgrades.compiledUpgradeFile === undefined) {
+      qs.logger.log(compileUpgradeResult.compiledUpgrades.scripts.map(s => s.description.description).join('\n'));
+    } else {
+      qs.logger.log(compileUpgradeResult.compiledUpgrades.descriptions.map(s => s.description).join('\n'));
+    }
```

Comment:
nit: Can you restructure the if expression? I find it hard to reason about. Something like this I find easier to read

```suggestion
    if (compileUpgradeResult.compiledUpgrades.compiledUpgradeFile) {
      // Single file upgrade
      qs.logger.log(compileUpgradeResult.compiledUpgrades.descriptions.map(s => s.description).join('\n'));
    } else {
      qs.logger.log(compileUpgradeResult.compiledUpgrades.scripts.map(s => s.description.description).join('\n'));
    }
```

Reply:
Is you suggestion mostly about switching the sense of the test or about the `=== undefined`.

It has to have the `=== undefined` part to get the types correct (according to the types we could have an empty `compiledUpgradeFile`). However I can flip the sense of the test if that makes more sense to you.

Reply:
I see why typing is broken if you avoid the `undefined`. I think just adding a comment would be sufficient for me.